generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/DateValueStep.java

```diff
@@ -5,7 +5,7 @@
 
 public class DateValueStep {
 
-    public static final String DATE_REGEX = "((\\d{4})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})))$";
+    public static final String DATE_REGEX = "((\\d{4,19})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})))$";
```

Comment:
Java supports (with the `u` denotation) years represented between 1 and 19 digits long. See [`DateTimeFormatterBuilder.appendValue`](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatterBuilder.html)

> maxWidth - the maximum field width of the printed field, from 1 to 19

Reply:
What about negative for BCE years?

Reply:
These aren't supported presently.

Reply:
Updated to restrict dates to current legal range